ui/src/Signin.tsx

```diff
@@ -74,13 +75,28 @@ export class Signin extends PureComponent<Props, State> {
   private checkForLogin = async () => {
     try {
       await client.users.me()
+      const redirectIsSet = !!window.localStorage.getItem('redirectTo')
+      if (redirectIsSet) {
+        window.localStorage.removeItem('redirectTo')
+      }
     } catch (error) {
       const {
         location: {pathname},
       } = this.props
 
       clearInterval(this.intervalID)
 
+      if (CLOUD && isFlagEnabled('redirectto')) {
+        const url = new URL(
+          `${window.location.origin}${CLOUD_SIGNIN_PATHNAME}?redirectTo=${
+            window.location.href
+          }`
+        )
+        window.localStorage.setItem('redirectTo', window.location.href)
+        window.location.href = url.href
+        throw error
```

Comment:
why `throw` the error if we're redirecting away from this page?

Reply:
The original implementation set this so I went based off of that. I think the understanding is that if there's an issue with the redirect, we will not want the user to continue on (since that'll navigate them to `/signin` which is reserved for CLOUD)